File: flutter_inappwebview_platform_interface/lib/src/types/content_blocker_trigger_resource_type.dart
import 'package:flutter_inappwebview_internal_annotations/flutter_inappwebview_internal_annotations.dart';

import '../content_blocker.dart';

part 'content_blocker_trigger_resource_type.g.dart';

///Class that represents the possible resource type defined for a [ContentBlockerTrigger].
@ExchangeableEnum()
class ContentBlockerTriggerResourceType_ {
  // ignore: unused_field
  final String _value;
  const ContentBlockerTriggerResourceType_._internal(this._value);

  static const DOCUMENT = ContentBlockerTriggerResourceType_._internal(
    'document',
  );
  static const IMAGE = ContentBlockerTriggerResourceType_._internal('image');
  static const STYLE_SHEET = ContentBlockerTriggerResourceType_._internal(
    'style-sheet',
  );
  static const SCRIPT = ContentBlockerTriggerResourceType_._internal('script');
  static const FONT = ContentBlockerTriggerResourceType_._internal('font');
  static const MEDIA = ContentBlockerTriggerResourceType_._internal('media');
  static const SVG_DOCUMENT = ContentBlockerTriggerResourceType_._internal(
    'svg-document',
  );

  ///Any untyped load
  static const RAW = ContentBlockerTriggerResourceType_._internal('raw');
}

File: flutter_inappwebview_android/android/src/main/kotlin/dev/nosferatu500/inappwebview/content_blocker/ContentBlockerTriggerResourceType.kt
package dev.nosferatu500.inappwebview.content_blocker

enum class ContentBlockerTriggerResourceType(private val value: String) {
  DOCUMENT("document"),
  IMAGE("image"),
  STYLE_SHEET("style-sheet"),
  SCRIPT("script"),
  FONT("font"),
  SVG_DOCUMENT("svg-document"),
  MEDIA("media"),
  POPUP("popup"),
  RAW("raw");

  fun equalsValue(otherValue: String): Boolean = value == otherValue

  override fun toString(): String = value

  companion object {
    @JvmStatic
    fun fromValue(value: String): ContentBlockerTriggerResourceType =
      entries.firstOrNull { value == it.value }
        ?: throw IllegalArgumentException("No enum constant: $value")
  }
}
